src/matcha_ml/state/matcha_state.py

```diff
@@ -80,6 +79,25 @@ class MatchaState:
 
     components: List[MatchaStateComponent]
 
+    def get_component(self, resource_name: str) -> Optional[MatchaStateComponent]:
+        """Get a component of the state given a resource name.
+
+        Args:
+            resource_name (str): the components resource name
+
+        Returns:
+            MatchaStateComponent: the state component matching the resource name parameter.
+        """
+        component = next(
+            filter(
+                lambda component: component.resource.name == resource_name,
+                self.components,
+            ),
+            None,
+        )
+
+        return component
```

Comment:
What happens if a command like this is being run:
```
self.get_component("cloud").find_property("resource-group-name").value
```
but the "cloud" component does not exist? Will the program carry on correctly?